k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java

```diff
@@ -0,0 +1,140 @@
+package com.fsck.k9.activity;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.mail.Message;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = "src/main/AndroidManifest.xml", sdk = 21)
+public class ActivityListenerTest {
+
+    private ActivityListener activityListener;
+    private Account account;
+    private String folder;
+    private String errorMessage;
+    private Message message;
+    private int count;
+
+    @Before
+    public void before() {
+        account = mock(Account.class);
+        when(account.getDescription()).thenReturn("account");
+        folder = "folder";
+        errorMessage = "errorMessage";
+        message = mock(Message.class);
+        count = 1;
+        activityListener = new ActivityListener();
+    }
+
+    @Test
+    public void folderStatusChanged__shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+        activityListener.folderStatusChanged(account, folder, count);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertEquals("Poll account:folder", operation);
+    }
+
+    @Test
+    public void synchronizeMailboxStarted__shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertEquals("Poll account:folder", operation);
+    }
+
+    @Test
+    public void synchronizeMailboxProgress_shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+        activityListener.synchronizeMailboxProgress(account, folder, count, count);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertEquals("Poll account:folder"+"\\u0020"+"1/1", operation);
```

Comment:
Looks like this is fixed in Robolectric 3.1.1 and no tests break with the new version. Please feel free to upgrade us.